sdk/core/azure-core/azure/core/_connection_string_parser.py

```diff
@@ -0,0 +1,31 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Dict
+
+CS_DELIMITER = ";"
+CS_VAL_SEPARATOR = "="
+
+def parse_connection_string_to_dict(conn_str):
```

Comment:
is this method exposed to users? I'm not quite sure having a generic parser in azure core provides much value -it'd prolly be easier for users to write a `.split(;)` for themselves than find this API.

Reply:
So I exposed it to users since it needs to be available across all the SDKs that use parsing. I wasn't sure if there was another way to make it available to all those SDKs without making it public.

So the original request for this parser came from #16782. In order to actually make it generic enough so that it can be used by all of the other parsers, we couldn't do specific key validation here. There are only a couple validations that are generic enough that they can be added here without really breaking any of the other parsers. For some SDK parsers, this does provide extra checks that weren't in place before, so I think there is some value there.